.github/workflows/release.yaml

```diff
@@ -67,20 +74,28 @@ jobs:
     - name: Sign images 
       uses: sigstore/cosign-installer@main
       with:
-        cosign-release: 'v0.2.0'
+        cosign-release: 'v1.4.1'
     
     # Use cosign to sign the images
-    - run: |
+    - env:
+        COSIGN_EXPERIMENTAL: "true"
+      run: |
         export KMS_VAL=gcpkms://projects/kaniko-project/locations/global/keyRings/cosign/cryptoKeys/cosign
-        cosign sign -kms $KMS_VAL gcr.io/kaniko-project/executor:${{ env.GITHUB_SHA }}
-        cosign sign -kms $KMS_VAL gcr.io/kaniko-project/executor:${{ steps.vars.outputs.tag }}
-        cosign sign -kms $KMS_VAL gcr.io/kaniko-project/executor:latest
+        cosign sign -kms $KMS_VAL gcr.io/kaniko-project/executor@${{ steps.build-and-push.outputs.digest }}
+        cosign sign gcr.io/kaniko-project/executor@${{ steps.build-and-push.outputs.digest }}
 
   build-debug:
+    permissions:
+      # Read the repo contents
+      contents: read
+      # Produce identity token for keyless signing
+      id-token: write
+
     env:
       GITHUB_SHA: ${{ github.sha }}
       GITHUB_REF: ${{ github.ref }}
```

Comment:
These seem to be redundant FYI: https://docs.github.com/en/actions/learn-github-actions/environment-variables